GameServerLib/Logic/Logger.cs

```diff
@@ -70,14 +75,71 @@ public void LogCoreError(string format, params object[] args)
             LogCoreError(string.Format(format, args));
         }
 
+        public void Flush()
+        {
+            _logWriter.Flush();
+        }
+
+        public void LogFatalError(string line)
+        {
+            Log(line, "FATAL_ERROR");
+            Flush();
+        }
+
+        public void LogFatalError(string format, params object[] args)
+        {
+            LogFatalError(string.Format(format, args));
+        }
+
         private class LogWriter
         {
             public string _logFileName;
             private object _locker = new object();
+            private StringBuilder _stringBuilder = new StringBuilder();
+            private const double REFRESH_RATE = 1000.0 / 10.0; //10fps
+            private System.Timers.Timer _refreshTimer;
 
             public LogWriter(string executingDirectory, string logFileName)
             {
                 CreateLogFile(executingDirectory, logFileName);
+
+                //Start refresh loop
+                _refreshTimer = new System.Timers.Timer(REFRESH_RATE)
+                {
+                    AutoReset = true,
+                    Enabled = true
+                };
+                _refreshTimer.Elapsed += (_, _2) => RefreshLoop();
+                _refreshTimer.Start();
+            }
+
+            public void Flush()
+            {
+                RefreshLoop();
+            }
+
+            private void RefreshLoop()
+            {
+                if (_stringBuilder.Length > 0)
+                {
+                    String text = _stringBuilder.ToString();
+                    _stringBuilder.Clear();
+                    Task.Factory.StartNew(() =>
+                    {
+                        Console.WriteLine(text);
```

Comment:
Change this to `Console.Write(text)` to avoid the issue linked to this PR earlier